examples/registration_form/lib/address/address_component.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:angular/angular.dart';
+import 'package:angular_components/angular_components.dart';
+import 'package:angular_forms/angular_forms.dart';
+
+@Component(
+    selector: 'address-cmp',
+    templateUrl: 'address_component.html',
+    exports: [
+      states
+    ],
+    directives: [
+      formDirectives,
+      materialInputDirectives,
+      MaterialAutoSuggestInputComponent,
+      RequiredFirstValidator,
+      RequiredState
+    ])
+class AddressComponent {}
+
+/// Validator that ensures that [requiredFirst] control has a value before this
+/// control.
+///
+/// For example an address input where given two address lines the second line
+/// shouldn't be the only one with a value.
+// TODO(alorenzen): Improve the interop between the two Controls. Errors should
+// clear when requiredFirst value is set.
```

Comment:
Just a note this may not be possible currently. I think it might create a circular DI dep since NgControl -> validators and you would need the control to reset the errors.

It might be a really nice feature if either:
a) A validator could have an optional method that gives the control instance when it is set on a control.
b) A validator's interface is expanded to be able to reset itself which gives a callback to the control.

One wrinkle of this is the validator is just one of many so probably all of the validators would need to be rerun on a reset call, or more state would need to be known or assumed to just clear certain errors.

The other way around this is to force validators to not be able to change their states. So the validator could throw if requiredFirst is set twice.